src/main/java/com/diffblue/deeptestutils/Reflector.java

```diff
@@ -406,20 +424,27 @@ private static void makePublic(final CtClass c) {
           implementation.setSuperclass(cl);
         }
 
-        // look for constructor
-        // create default constructor if none exists
-        boolean foundDefault = false;
+        // In the case of an abstract class, search any constructor in the
+        // superclass and add an empty one with the same types
         if (!cl.isInterface()) {
+          boolean found = false;
```

Comment:
Since you're not actually looping over the constructors why not:

```
constructors = cl.getConstructors();
if(constructors.size() > 0) {
  CtConstructor newCtor = 
    CtNewConstructor.make(
      constructors.get(0).getParameterTypes(), ...
  // ...
} else {
   throw new NotFoundException(
}
```

I think it makes it more obvious what you are doing.